pennylane/math/multi_dispatch.py

```diff
@@ -315,6 +315,44 @@ def matmul(tensor1, tensor2, like=None):
     return ar.numpy.matmul(tensor1, tensor2, like=like)
 
 
+@multi_dispatch(argnum=[0, 1])

```

Comment:
With each new function added, we should include a test in `test/math/test_multi_dispactch`